src/people.cpp

```diff
@@ -23,12 +23,28 @@
 Guests _guests; ///< %Guests in the world/park.
 Staff _staff;   ///< %Staff in the world/park.
 
-static const uint32 COMPLAINT_TIMEOUT  = 8 * 60 * 1000;  ///< Time in milliseconds between two complaint notifications of the same type.
-static const uint16 COMPLAINT_THRESHOLD_HUNGER    = 80;  ///< After how many hunger    complaints a notification is sent.
-static const uint16 COMPLAINT_THRESHOLD_THIRST    = 80;  ///< After how many thirst    complaints a notification is sent.
-static const uint16 COMPLAINT_THRESHOLD_WASTE     = 30;  ///< After how many waste     complaints a notification is sent.
-static const uint16 COMPLAINT_THRESHOLD_LITTER    = 25;  ///< After how many litter    complaints a notification is sent.
-static const uint16 COMPLAINT_THRESHOLD_VANDALISM = 15;  ///< After how many vandalism complaints a notification is sent.
+static const uint32 COMPLAINT_TIMEOUT = 8 * 60 * 1000;  ///< Time in milliseconds between two complaint notifications of the same type.
+static const uint16 COMPLAINT_THRESHOLD[Guests::COMPLAINT_COUNT] = {  // Indexed by %Guests::ComplaintType.
+		80,  ///< After how many hunger    complaints a notification is sent.
+		80,  ///< After how many thirst    complaints a notification is sent.
+		30,  ///< After how many waste     complaints a notification is sent.
+		25,  ///< After how many litter    complaints a notification is sent.
+		15,  ///< After how many vandalism complaints a notification is sent.
+};
+static const StringID COMPLAINT_MESSAGES[Guests::COMPLAINT_COUNT] = {  // Indexed by %Guests::ComplaintType.
+		GUI_MESSAGE_COMPLAIN_HUNGRY,     ///< Message for complaints about lack of food.
+		GUI_MESSAGE_COMPLAIN_THIRSTY,    ///< Message for complaints about lack of drink.
+		GUI_MESSAGE_COMPLAIN_TOILET,     ///< Message for complaints about lack of toilets.
+		GUI_MESSAGE_COMPLAIN_LITTER,     ///< Message for complaints about dirty paths.
+		GUI_MESSAGE_COMPLAIN_VANDALISM,  ///< Message for complaints about demolished objects.
+};
```

Comment:
By specifying the size of the array, the compiler will add null entries to extend the array if needed, and you're not warned about it. As those entries probably won't do much useful, you can create bugs by adding new complaints but forgetting to add a line here.

A tighter check can be done with `assert_compile()`, which can be used outside functions, and simply refuses to compile if the assertion fails.

If I remember correctly, if you remove the `Guests::COMPLAINT_COUNT` text, and thus give the compiler room to only make space for the entries that are listed, and add below the closing curly brace add a line `assert_compile(lengthof(COMPLAINT_MESSAGES) == Guests::COMPLAINT_COUNT);`
then the number of entries in the array must match the value of the enumeration or it won't compile.

A simple test is to just comment out one of the lines in the array and trying to compile the code.
I haven't tried it (don't think my dev environment works any more), but it should work.

One small note about `lengthof`, it needs a fully defined array as parameter (that is, not a declaration without having seen a proper definition), or you get nonsense values.

Reply:
Done

Thanks for the review :)